src/middlewared/middlewared/plugins/enclosure_/ses_enclosures2.py

```diff
@@ -0,0 +1,27 @@
+from contextlib import suppress
+from logging import getLogger
+from pathlib import Path
+
+from libsg3.ses import EnclosureDevice
+from .enclosure_class import Enclosure
+
+logger = getLogger(__name__)
+
+
+def get_ses_enclosure_status(bsg_path):
+    try:
+        return EnclosureDevice(bsg_path).status()
+    except OSError:
+        logger.error('Error querying enclosure status for %r', bsg_path, exc_info=True)
+
+
+def get_ses_enclosures(dmi):
+    rv = list()
+    with suppress(FileNotFoundError):
```

Comment:
Should we be suppressing FileNotFoundError here or below where we get the enclosure status? Do we want to break iteration early?